custom-pages/login/src/routes/Login/index.test.js

```diff
@@ -1,15 +1,56 @@
 import React from 'react'
 import Enzyme, { shallow } from 'enzyme'
 import Adapter from 'enzyme-adapter-react-14'
-
 import Login from './'
 
 Enzyme.configure({ adapter: new Adapter() })
 
 describe('Login components', () => {
-  jest.mock('../../services/AuthApi', () => jest.fn(() => {}))
+  let el
+  let instance
+  const auth = {
+    login: jest.fn()
+  }
+  const functionSignature = 'a function signature'
+  jest.mock('../../services/AuthApi')
+
+  beforeEach(() => {
+    el = shallow(<Login.component />)
+    instance = el.instance()
+    instance.auth = auth
+    instance.requestErrorCallback = functionSignature
+  })
 
   it('should render <Login /> correctly', () => {
-    expect(shallow(<Login.component />)).toMatchSnapshot()
+    expect(el).toMatchSnapshot()
+  })
+
+  it('should instantiate AuthApi', () => {
+    expect(instance.auth).toBe(auth)
+  })
+
+  it('should render correctly when there is an error', () => {
+    el.setState({ error: 'this is an error' }).update()
+    expect(el).toMatchSnapshot()
+  })
+
+  it('should render correctly when there is username and password', () => {
+    el.setState({
+      valid: true
+    }).update()
+    expect(el).toMatchSnapshot()
+  })
+
+  it('should call the AuthApi login when login is invoked', () => {
+    const username = 'username'
+    const password = 'pwd'
+
+    el.setState({
+      username,
+      password
+    }).update()
+
+    instance.login()
+    expect(instance.auth.login).toBeCalledWith(username, password, functionSignature)
```

Comment:
there don't seem to be any form validation tests here. can they be added please?

Reply:
there is no form validation on Login. what validation should there be?